src/main/scala/ir/IRCursor.scala

```diff
@@ -148,27 +148,21 @@ object IntraProcBlockIRCursor extends IntraProcBlockIRCursor
 trait InterProcIRCursor extends IRWalk[CFGPosition, CFGPosition] {
 
   final def succ(pos: CFGPosition): Set[CFGPosition] = {
-    IntraProcIRCursor.succ(pos) ++
-      (pos match
+      pos match
         case c: DirectCall if c.target.blocks.nonEmpty => Set(c.target)
         case c: Return => c.parent.parent.incomingCalls().map(_.successor).toSet
-        case _         => Set.empty
-      )
+        case _         => IntraProcIRCursor.succ(pos)
   }
 
   final def pred(pos: CFGPosition): Set[CFGPosition] = {
-    IntraProcIRCursor.pred(pos) ++
-      (pos match
-        case c: Command => {
+      pos match
+        case c: Command =>
           IRWalk.prevCommandInBlock(c) match {
             case Some(d: DirectCall) if d.target.blocks.nonEmpty => d.target.returnBlock.toSet
-            case o            => o.toSet
+            case o            => o.toSet ++ IntraProcIRCursor.pred(pos)
           }
-
-        }
-        case c: Procedure                              => c.incomingCalls().toSet.asInstanceOf[Set[CFGPosition]]
-        case _                                         => Set.empty
-      )
+        case c: Procedure => c.incomingCalls().toSet.asInstanceOf[Set[CFGPosition]]
+        case _            => IntraProcIRCursor.pred(pos)
   }
 }
```

Comment:
Changing the behaviour of InterProcIRCursor may break other analyses that rely on it.

Reply:
It was not producing the expected behaviour. It was returning the edge from the call to the after call which is not correct